tests/test_pipelines.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+
+from pytest_twisted import ensureDeferred
+from scrapy import Spider
+from scrapy.crawler import Crawler
+from scrapy.utils.test import get_crawler
+
+
+@ensureDeferred
+async def test_duplicate_url_discarder_pipeline_no_addon(caplog, httpserver) -> None:
+    caplog.set_level("INFO")
+
+    httpserver.expect_request("")
+    start_url = httpserver.url_for("")
+
+    class FakeSpider(Spider):
+        name = "fake_spider"
+        start_urls = [start_url]
```

Comment:
If you don’t care about the HTTP handled being used, you could use `data:,` instead, no need for an HTTP server, and the test should be faster.

Reply:
TIL. Thanks super handy. Thanks for raising it! https://github.com/zytedata/duplicate-url-discarder/pull/17/commits/c12a803bb85578662a639f2ec68968d2123c8574